src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/rest/RestKNNStatsHandler.java

```diff
@@ -73,17 +75,35 @@ public String getName() {
         return NAME;
     }
 
+
+    private List<String> getStatsPath() {
+        List<String> statsPath = new ArrayList<>();
+        statsPath.add("/{nodeId}/stats/");
+        statsPath.add("/{nodeId}/stats/{stat}");
+        statsPath.add("/stats/");
+        statsPath.add("/stats/{stat}");
+        return statsPath;
+    }
+
+    private Map<String, String> getNewURLPathByLegacyURLPath() {
+        return getStatsPath().stream().collect(
+            Collectors.toMap(path -> KNNPlugin.LEGACY_KNN_BASE_URI + path, path -> KNNPlugin.KNN_BASE_URI + path)
+        );
+    }
+
     @Override
     public List<Route> routes() {
-        return ImmutableList.of(
-                new Route(RestRequest.Method.GET, KNNPlugin.KNN_BASE_URI + "/{nodeId}/stats/"),
-                new Route(RestRequest.Method.GET, KNNPlugin.KNN_BASE_URI + "/{nodeId}/stats/{stat}"),
-                new Route(RestRequest.Method.GET, KNNPlugin.KNN_BASE_URI + "/stats/"),
-                new Route(RestRequest.Method.GET, KNNPlugin.KNN_BASE_URI + "/stats/{stat}")
-        );
+      return ImmutableList.of();
```

Comment:
Why is this empty?

Reply:
I had the same question, is that because `replacedRoutes` is sufficient?